apollo-router/src/plugins/connectors/handle_responses.rs

```diff
@@ -48,116 +50,137 @@ pub(crate) async fn handle_responses(
     let count = responses.len();
 
     for response in responses {
```

Comment:
Small nit: you could deconstruct the tuple (or separate struct) here to make all uses of `response.#` a little more legible. Something like the following:

```rust
for (fetch_result, response_key) in responses {...}
```